### ISSUE TITLE
Collaboration Syncplay 1.6.4a - VLC 3.0.8 on MAC shows problem

### ISSUE THREAD

**Issue by Wol-Loth** [Syncplay/syncplay#302]
I'm new in this forum, so be kind and help me solve my problem 🙏
To synchronise external sound for Zoom-Meeting I wanted to use Syncplay in combination of VLC, both latest versions  1.6.4a resp. 3.0.8, on my MacBook Pro Mid 15 (MAC OS 10.14.6) and at friends with Mac and Win PCs.
I use VLC for long time and use actual updates. Syncplay was downloaded from Syncplay.pl homepage and installed as recommended by drag-drop the button on the programs button.
Trying to run Syncplay  opened the settings windows and asked for the file source.
When giving a local mp4-file and starting Syncplay connects to the desired server, starts the VLC app and the file is processed, but after short time two error messages appear (see screenshots) and no commands are taken any more.
In the documentation I found that Syncplay works with VLC 2.X. 
Can anyone give me a hint how to run Syncplay 1.6.4a and VLC 3.0.8 for synchronisation over internet ?

Regards  Wolfgang

<img width="708" alt="Screenshot 2020-04-22 14 35 37" src="https://user-images.githubusercontent.com/64156487/80011125-b6e65f80-84cb-11ea-928c-9d83bcedc0ee.png">
<img width="704" alt="Screenshot 2020-04-22 14 36 10" src="https://user-images.githubusercontent.com/64156487/80011134-b8b02300-84cb-11ea-9425-74a0f01748d4.png">

**Comment by albertosottile** [comment]
Hello, thanks for reporting this. I am not completely sure that I understand your use case.

Anyway, I would check if the helper LUA script was copied correctly in the right location by the app. Could you check the content of the folder `/Users/wolfgangloth/Library/Application Support/org.videolan.vlc/lua/intf` and post it here? You can use either the Terminal or the Finder Go to Folder menu to access this folder.

**Comment by Wol-Loth** [comment]
THX for the quick reply.
I checked for the path but in Application support is no folder "org.videolan.vlc/lua/intf"
I've checked for vlc.lua and found it  in 
wolfgangloth/Applications/ VLC/ Contents/MacOS/share/lua/inf/syncplay.lua
syncplay.lua size is 20 kb
Is this the wrong place ?

**Comment by albertosottile** [comment]
Interesting... Could you take a screenshot of the first dialog that opens when you start Syncplay (the Syncplay Configuration window)? Thanks

**Comment by Wol-Loth** [comment]
<img width="812" alt="Screenshot 2020-04-22 19 55 13" src="https://user-images.githubusercontent.com/64156487/80016330-42afba00-84d3-11ea-9c36-1f3f87128b09.png">

**Comment by albertosottile** [comment]
Ok, it seems that you have two instances of VLC installed in your system, one in `/Applications/VLC.app` (the one recognized by Syncplay) and another in `/Users/wolfgangloth/Applications/VLC.app`. Apparently, by default, Syncplay opens VLC from the former path but tries to copy the `syncplay.lua` script in the other path. 

I would just delete one of the two VLC apps, if possible. Otherwise, you can select the copy installed in your home folder from the Syncplay configuration dialog (Path to media player) and try to run Syncplay in any case.

**Comment by Wol-Loth** [comment]
So in the /Applications/VLC.app path there is no syncplay.lua ; but all modules end with ".luac"
Should I transfer the synplay.lua to this place and delete the other path ?
<img width="992" alt="Screenshot 2020-04-22 20 41 09" src="https://user-images.githubusercontent.com/64156487/80020845-f61bad00-84d9-11ea-9675-fb19155a0d0a.png">

**Comment by Wol-Loth** [comment]
I have renamed the second path from "Application Support" and copied the syncplay.lua to the "Applications" path ... but got the same results.
One strange thing: When I downloaded  syncplay.lua it appeared in the download folder as "syncplay.lua.html". I guessed that this was a download failure and renamed it to syncplay.lua.
Was this a mistake ????

**Comment by albertosottile** [comment]
I would recommend deleting both the VLC apps from `/Applications` and `/Users/wolfgangloth/Applications/`, then reinstalling VLC, then trying again with Syncplay.

**Comment by Wol-Loth** [comment]
I have reinstalled both, but can't find syncplay.lua in my vlc.app container !
As I understand it should be installed automatically to VLC.app, right ?
I have found it and copied it to VLC.app , ... but running syncplay gives the same results !
" Can't find syncplay.lua "

**Comment by Et0h** [comment]
Did you reinstall both instances of VLC? That could be part of the problem.

**Comment by albertosottile** [comment]
I would just download the `syncplay.lua` file from here https://raw.githubusercontent.com/Syncplay/syncplay/master/syncplay/resources/lua/intf/syncplay.lua (be careful with the extension) and put it here `/Users/wolfgangloth/Library/Application Support/org.videolan.vlc/lua/intf`. Create all the subfolders needed if they are not there.

**Comment by Wol-Loth** [comment]
So sorry, it doesn't work !
What does the message mean ?
syncplay.lua: 542 : Cannot listen on localhost: 29071

Is there some other player for MAC and WIN you propose that will not give this problem ?

**Comment by Wol-Loth** [comment]
Seems to me as if there is only one way connection syncplay --> vlc , but the return path is missing !
So Syncplay starts VLC, i can hear the music and see the VLC window and Syncplay says at the same time in the right up window "no file being played" , i.e. the listening channel of Syncplay is not assigned or wrong assigned  ???

**Comment by Wol-Loth** [comment]
Alberto, I have uninstalled VLC completely.  Downloaded and installed VLC new.
Running Syncplay always gives me the same error messages
... and each start of Syncplay starts a new VLC session shown as icon in the task bar.
And ... I can't drag a file to the "share list window"
<img width="275" alt="Screenshot 2020-04-25 15 32 18" src="https://user-images.githubusercontent.com/64156487/80281314-89d3c000-870a-11ea-808c-f492e49785fd.png">
<img width="1128" alt="Screenshot 2020-04-25 15 33 23" src="https://user-images.githubusercontent.com/64156487/80281315-8b9d8380-870a-11ea-927a-198024f1dcb0.png">
<img width="191" alt="Screenshot 2020-04-25 15 34 08" src="https://user-images.githubusercontent.com/64156487/80281318-8e987400-870a-11ea-972b-11e6c2ad4436.png">
<img width="1154" alt="Screenshot 2020-04-25 15 34 29" src="https://user-images.githubusercontent.com/64156487/80281319-8e987400-870a-11ea-9c86-56b5ca1d2d0f.png">

I was so full of hope and joy giving my chorus some possibility to sing together even though they can't hear it simultaneously because of the internet delay.
Don't have any idea how to get it work ?  Please 🙏

**Comment by daniel-123** [comment]
mpv has macOS release: https://mpv.io/installation/

Though I don't have any macOS device or experience to tell how well it works there. As far as Syncplay goes it's the player that integrates very nicely without need for any extra layers (which are needed for VLC).

**Comment by Wol-Loth** [comment]
Thank you Daniel, I read about MPV in the syncplay docu. 
But I i'm out of programing (FORTRAN) since 30 years - odd user 😔
and doesn't know how to install it .

**Comment by albertosottile** [comment]
There's no need to be a programmer for installing mpv, you can get the latest app from https://laboratory.stolendata.net/~djinn/mpv_osx/mpv-latest.tar.gz . Just open the archive, move the mpv app it in `/Applications` and open it once. Then close it, try to open Syncplay and select mpv from the `Path to media player` field in the configuration dialog.

**Comment by Wol-Loth** [comment]
Thanks a lot Alberto 🙏
This works w/o error messages 👍
The Win version is as simple to install at my colleagues ?
Which Win installation would you prefer/propose ?

**Comment by Et0h** [comment]
In line with the suggestion from @bobismijnnaam I have now now made commit ff6a6ad5a50180838fbd84c8f110d0b11495ff87 which will hopefully fix everything. Please try out the latest release from https://bintray.com/syncplay/Syncplay/Syncplay/v1.6.5#files and let me know if that fixes VLC support.

In terms of mpv on Windows, I personally use the latest stable releases from https://sourceforge.net/projects/mpv-player-windows/files/stable/ - for now Syncplay only supports up to mpv 0.32.0 due to a breaking change, so avoid the GIT HEAD releases for the time being. you may need to get the 7-Zip utility to extract the files. x86_64 is for 64-bit versions of Windows and  i686 means is for 32-bit versions of Windows.

**Comment by Wol-Loth** [comment]
EtOh, so sorry ... gives the same error message as 1.6.4a to me 😢
I deleted SyncPlay from "Programs" and installed new version via dmg-file.

But today we had my first synchron trial playing a "mp4" via WIN - SP / VLC and MAC - SP / mpv and it  worked as expected 🤩
Hope that we get MAC-SP / VLC also running, cause it is the most common player for my friends and already on their PCs.
Thank you very much for your commitment 🙏
If I can do something to help you find and eleminate the bug pls let me know !
<img width="1280" alt="Screenshot 2020-05-02 23 19 10" src="https://user-images.githubusercontent.com/64156487/80892728-a2287980-8ccc-11ea-8f1a-44d1e94d5811.png">

**Comment by Et0h** [comment]
It says it can't listen on a specific port. Is this a permission issue, e.g. you need to run it as administrator or change your firewall rules?

**Comment by Wol-Loth** [comment]
But the port number changes with every Syncplay start  o-) ! 
Tried to ping the localhost, but " cannot resolve localhost ... unknown host !"
I'm running the MacBook as Admin !

**Comment by Wol-Loth** [comment]
Hi friends, got a new problem : 
When playing mp3  on MAC 10.14.6 (syncplay 1.6.5 / mpv) and WIN10 (Syncplay 1.6.4a /VLC) we could try to start the file playing, but my system always stepped back some seconds,  played 3 sec stepped back etc.  w/o touching the mpv-window ! ??? How can I avoid this ? mp4 is fine !
Sorry, forgot screenshot 🙏

**Comment by Wol-Loth** [comment]
Tried solo playing "mp3" on MAC and found the same situation ! With screenshot !
What to do ???
<img width="701" alt="Screenshot 2020-05-03 15 59 37" src="https://user-images.githubusercontent.com/64156487/80916222-a057c800-8d57-11ea-9ab0-c28ccbe383c6.png">

**Comment by Et0h** [comment]
@Wol-Loth The latest build at https://bintray.com/syncplay/Syncplay/Syncplay/v1.6.5#files should hopefully fix playing music files on mpv. Please let me know how if it works!

**Comment by Wol-Loth** [comment]
@Et0h , SyncPlay-mpv on MAC-OS with mp3 and mp4 works well, thanks a lot 🙏 !
... the connection to VLC on MAC is still not working see screenshot
<img width="694" alt="Screenshot 2020-05-17 22 08 58" src="https://user-images.githubusercontent.com/64156487/82158986-9cfe2980-988b-11ea-98ed-e8e1f589583c.png">

**Comment by Et0h** [comment]
Glad to hear mpv is working. In terms of VLC the issue seems to be on the VLC side, not the Syncplay side. VLC is saying it can't listen on a port, but it isn't saying why. It could for example be a permission issue (is it being run as administrator/root?) or a firewall issue.

**Comment by Et0h** [comment]
I seem to recall that on Linux if you are running Syncplay as root then you had to run VLC via vlc-wrapper to drop the privileges to allow access to listening on a socket. There might be a similar issue for macOS.

**Comment by Wol-Loth** [comment]
I run the MACBook as admin and don't think it's a permission problem, 'cause the "port number" changes with every  program start. How to give a permission, when the port changes ?

**Comment by Wol-Loth** [comment]
I'll put the question in the VideoLAN-Forum, let's wait and see 😉

**Comment by Et0h** [comment]
For anyone wanting to keep track, the relevant thread on the VLC forums is: https://forum.videolan.org/viewtopic.php?f=29&t=111639&sid=dd3c4dab81e39bad07aabd448774024a

**Comment by Wol-Loth** [comment]
@Et0h : What is the Windows Mobil version good for ?
Is it for Windows-Laptops or for Mobile Pads with WIN ?
... and is there a "mobil" version for iPads underway ?
and   ....
Just learned in a ZOOM meeting that the media player with WIN 10 is mp2 and so seems compatible with SyncPlay  she told me :o)
Have to check it with the participant tomorrow and tell you the result (on an other thread ?)

**Comment by Et0h** [comment]
@Wol-Loth What Windows Mobile version? We have only ever released Syncplay for PCs (inc. Windows, Mac, Linux and *BSD).

There are no plans for any official releases on other platforms, but Syncplay is open source so anyone is free to create their own unofficial iPad version if they wished (so long as they don't pretend it is an official Syncplay release).

**Comment by albertosottile** [comment]
Closing this as we did not manage to identify (let alone reproduce) the underlying bug. Please feel free to reopen this anytime.

**Comment by Et0h** [comment]
Thanks for reporting the issue @smiba 

We have looked into this issue and have not found a solution so please continue to sue mpv for now. I have now updated the troubleshooting page at https://syncplay.pl/guide/trouble/ to point them to this issue and advise that they use mpv.

If any developer can figure out a good fix for this then please make a pull request, but it is being closed for now as we do not have a fix for it.

**Comment by smiba** [comment]
Hi, I've fixed the issue! I had a broken /etc/hosts file, causing localhost to not resolve to anything

For some reason the line endings had been removed causing the default of:
```
##
# Host Database
#
# localhost is used to configure the loopback interface
# when the system is booting.  Do not change this entry.
##
127.0.0.1 localhost
255.255.255.255 broadcasthost
::1 localhost
```

To morph into:
```
## # Host Database # # localhost is used to configure the loopback interface # when the system is booting.  Do not change this entry. ## 127.0.0.1 localhost 255.255.255.255 broadcasthost ::1 localhost
```

Because of this default definitions like localhost did not exist on my system (which probably caused some other issues too!) 

This can be fixed in syncplay by referring to 127.0.0.1 in the script, but this is a system configuration error primarily. People can check if this is the issue they're running into by trying to ping or resolve localhost
